### ISSUE TITLE
Core assignation gets written in Playlists

### ISSUE THREAD

**Issue by Tatsuya79** [libretro/RetroArch#2959]
If I've got a playlist for Wonderswan games, first game is like this:

..\..\Roms\Bandai - WonderSwan\WonderSwan Color\Alchemist Marie & Elie - Futari no Atelier (Japan).zip#Alchemist Marie & Elie - Futari no Atelier (Japan).wsc
Alchemist Marie & Elie - Futari no Atelier (Japan)
DETECT
DETECT
B18CDC0D|crc
Bandai - WonderSwan Color.lpl

If I went into settings -> Playlists and associated that playlist with mednafen_wswan, it will become like this:

..\..\Roms\Bandai - WonderSwan\WonderSwan Color\Alchemist Marie & Elie - Futari no Atelier (Japan).zip#Alchemist Marie & Elie - Futari no Atelier (Japan).wsc
Alchemist Marie & Elie - Futari no Atelier (Japan)
E:\Emulateurs\RetroArch64\cores\mednafen_wswan_libretro.dll
WonderSwan/Color (Mednafen Cygne)
B18CDC0D|crc
Bandai - WonderSwan Color.lpl

Not sure why it happens.
If I keep the DETECT lines unaltered by setting my playlist as read-only, it launches without problem.

That's bad for portability and if you want to change the core assignation (what's written in the playlist gets priority over the setting->Playlists assignation).

**Comment by Kivutar** [comment]
It was like this because some people want to keep memory of "always launch this game with this core, and this game with this other core".

**Comment by Tatsuya79** [comment]
I'm not sure that's the best behaviour, as playlists don't stay clean.

Perhaps make it write to the playlist only when core assignation setting is on N/A?
Or just manually.

**Comment by Kivutar** [comment]
I agree with you. The best behaviour would be to not write it when a default core is set. I tried to fix it some month ago and failed. It requires some refactoring.

**Comment by ofry** [comment]
I like if this feature will add.

**Comment by fr500** [comment]
https://github.com/libretro/RetroArch/issues/3558
https://github.com/libretro/RetroArch/issues/3456

**Comment by fr500** [comment]
https://github.com/libretro/RetroArch/issues/2064

**Comment by zoltanvb** [comment]
This is solved at least since 0be44e5, playlist default core is a separate entry and does not rewrite individual selections.
